### DISCUSSION TITLE
Windows 10 -> 11 : Warning LGHT1076 : ICE91

### DISCUSSION THREAD

**Discussion by Kowolik10** [wixtoolset/issues#8680] [Questions]
Hello,

We are using the wix created .msi installer for our product.
The install scope of our product is set to "perMachine" but we also need to copy some files to the current user by accesing the DirectoryRef Id="LocalAppDataFolder".

It was working fine on windows 10 but now after we migrated to windows 11 files are not copied to the LocalAppDataFolder.

The output says:

**The file "..." will be installed to the per user directory 'folder-name' that doesn't vary based on ALLUSERS value. This file won't be copied to each user's profile even if a per machine installation is desired.**

warning LGHT1076 : ICE91: 

Is there anything we can do to archieve our goal ?

**Comment by bevanweiss** [comment]
LocalAppDataFolder is a bad idea if the installer is being run elevated (like with perMachine).
The elevation may redirect this default windows user folder.  Since the InstallFiles will be run as LocalSystem
The warning will have been in the whole time, you probably shouldn't have ignored it.  It's specifically telling you that you're going against a Microsoft guideline for installers.. and hence they may decide to break your unsupported operations in the future (as it looks like they have done between W10 and W11)...  you play the risky game, you take the risky results....


You could use a SetProperty to assign the current (unelevated) user folder to a property that is then passed into the elevated execution.
This action would also need to be performed before elevating..
This would then be the directory location for the file copy.  You might want to flag these files as permanent though, so uninstalls don't fail if they can't be accessed..

Although the premise of the question sounds wrong anyway.
If performing a Per Machine installation, WHY should any files be copied into only a single users directory?
Surely all users have access to the remainder of the installation (executables etc etc), why do they not also get these files?
Why would other computer administrators not get copies of these files?

What if another administrator performs the uninstall, but doesn't have permissions to access the original installers user directory?
